src/commands/Social/remindme.ts

```diff
@@ -68,46 +68,48 @@ export default class extends SkyraCommand {
 		return message.sendLocale('COMMAND_REMINDME_DELETE', [selectedTask]);
 	}
 
-	public async parseInput(message: KlasaMessage, input: string) {
-		const parsed: { time: number | null; title: string | null } = {
-			time: null,
-			title: null
-		};
-
-		if (/^in\s/.test(input)) {
-			const indexOfTitle = input.lastIndexOf(' to ');
-			parsed.time = new Duration(input.slice(3, indexOfTitle > -1 ? indexOfTitle : undefined)).offset;
-			parsed.title = indexOfTitle > -1 ? input.slice(indexOfTitle + 4) : 'Something, you did not tell me what to remind you.';
-		} else {
-			const indexOfTime = input.lastIndexOf(' in ');
-			parsed.title = input.slice(/^to\s/.test(input) ? 3 : 0, indexOfTime > -1 ? indexOfTime : undefined);
-
-			if (indexOfTime !== -1) {
-				parsed.time = new Duration(input.slice(indexOfTime + 4)).offset;
+	public async init() {
+		this.createCustomResolver('action', (arg, _possible, message) => {
+			if (!arg) return Actions.List;
+
+			switch (arg.toLowerCase()) {
+				case 'a':
+				case 'all':
+				case 'l':
+				case 'list': return Actions.List;
+				case 'r':
+				case 'rm':
+				case 'remove':
+				case 'd':
+				case 'del':
+				case 'delete': return Actions.Delete;
+				case 'c':
+				case 'create': message.args.splice(message.params.length, 0, undefined!);
+				// Fallback
+				case 'me':
+				default: return Actions.Create;
 			}
-		}
-
-		if (!util.isNumber(parsed.time) || parsed.time < 59500 || parsed.time > (Time.Year * 5)) {
-			parsed.time = await this.askTime(message, message.language.tget('COMMAND_REMINDME_INPUT_PROMPT'));
-		}
-
-		return parsed;
-	}
-
-	public async askTime(message: KlasaMessage, alert: string) {
-		await message.sendMessage(alert);
+		});
 
-		let time: number;
-		let attempts = 0;
-		do {
-			const messages = await message.channel.awaitMessages(msg => msg.author === message.author, { time: 30000, max: 1 });
-			if (!messages.size) throw null;
-			time = new Duration(messages.first()!.content).offset;
-			attempts++;
-		} while (time < 60000 && attempts < 5);
+		this.createCustomResolver('idOrDuration', (arg, possible, message, [action]: Actions[]) => {
```

Comment:
The type for the action destructuring should be `[Actions]`

Reply:
Not possible, it complains that it doesn't match `any[]`.